cloudinit/sources/DataSourceVMware.py

```diff
@@ -12,6 +13,7 @@
 This module provides a cloud-init datasource for VMware systems and supports
 multiple transports types, including:
 
+    * VMware-imc (VMware Guest Customization)
```

Comment:
EnvVars should _always_ be the first check as they are there for using this DataSource in a container. The order should be:

1. EnvVars
2. VMware-imc
3. GuestInfo

And actually I think the order should _likely_ put VMware-imc at the bottom as it is new to the order. The other checks are _really_ fast, so I am concerned about the time it will take to move past the new imc check.

Also, I know John S. would like us to go ahead and slot in support for DataSets, even if they are not yet totally ready. Should we do that with this PR or in a follow-up?

Reply:
Due to the OVF DataSource is listed before VMware DataSource in search list, and VMware-imc is moving from OVF to VMware, in order to maintain backward compatibility, so I put VMware-imc at the first place.
While we are putting these transports includes future DataSets into our own VMware DataSource, we can re-arrange them, for now I don't think people will set both VMware-imc and GuestInfo data together, I will update this part.
For the DataSets, we can add it in a future PR.